src/settings/components/Period/Period.test.js

```diff
@@ -0,0 +1,195 @@
+    const changeFormValueMock = jest.fn();
+
+    render(<Period {...commonMockedProps} changeFormValue={changeFormValueMock} />);
+
+    const finePolicyDurationColumn = screen.getByTestId('finePolicyDurationColumn');
+    const textFieldOnClearFieldTestElement =
+      within(finePolicyDurationColumn).getByTestId('onClearFieldTestElement');
+
+    fireEvent.click(textFieldOnClearFieldTestElement);
+
+    expect(changeFormValueMock).toHaveBeenCalledWith(commonMockedProps.inputValuePath, '');
+  });
+
+  it('should render component when "selectPlaceholder" props is passed', () => {
+    const testSelectPlaceholder = 'testSelectPlaceholder';
+
+    render(<Period {...commonMockedProps} selectPlaceholder={testSelectPlaceholder} />);
+
+    expect(Field).toHaveBeenNthCalledWith(2, expect.objectContaining({
+      placeholder: testSelectPlaceholder,
+    }), {});
+  });
+
+  it('should render component when "fieldLabel" props is passed', () => {
+    const mockedProps = {
+      ...commonMockedProps,
+      fieldLabel: labelIds.fieldLabelId,
+    };
+
+    render(<Period {...mockedProps} />);
+
+    const periodLabelSection = screen.getByTestId('periodLabelSection');
+    expect(periodLabelSection).toBeVisible();
+
+    const periodLabel = within(periodLabelSection).getByText(labelIds.fieldLabelId);
+    expect(periodLabel).toBeVisible();
+  });
+
+  it('should render component when "required" props is passed', () => {
+    const mockedProps = {
+      ...commonMockedProps,
+      fieldLabel: labelIds.fieldLabelId,
+      required: true,
+    };
+
+    render(<Period {...mockedProps} />);
+
+    const periodLabelSection = screen.getByTestId('periodLabelSection');
+    const periodLabel = within(periodLabelSection).getByText(labelIds.fieldLabelId);
+    expect(periodLabel).toHaveAttribute('required');
+  });
+
+  it('should render component when "inputSize" props is passed', () => {
+    const mockedProps = {
+      ...commonMockedProps,
+      inputSize: 3,
+    };
+
+    render(<Period {...mockedProps} />);
+
+    const finePolicyDurationColumn = screen.getByTestId('finePolicyDurationColumn');
+    expect(finePolicyDurationColumn).toHaveAttribute('xs', '3');
+  });
+
+  it('should render component when "selectSize" props is passed', () => {
+    const mockedProps = {
+      ...commonMockedProps,
+      selectSize: 3,
+    };
+
+    const { rerender } = render(<Period {...mockedProps} />);
+
+    const finePolicyIntervalColumn = screen.getByTestId('finePolicyIntervalColumn');
+    expect(finePolicyIntervalColumn).toHaveAttribute('xs', '3');
+
+    rerender(
+      <Period
+        {...mockedProps}
+        intervalPeriodsSuffix={labelIds.intervalPeriodsSuffixId}
+      />
+    );
+
+    const periodIntervalColumn = screen.getByTestId('periodIntervalColumn');
+    expect(periodIntervalColumn).toHaveAttribute('xs', '3');
+  });
+
+  it('should render component when "intervalPeriodsSuffix" props is passed', () => {
```

Comment:
we can swap this test and the previous one. in this case we first check correct render `intervalPeriodsSuffix` section when corresponding property is passed, and then we can render component with 'intervalPeriodsSuffix' prop, and with 'selectSize' which is differ from default. so we can avoid step with rerender and just check that both sections have correct 'xs' attribute.

Reply:
done. thanks